### PR TITLE
Added `max_concurrency` for bulk requests

### PR DESCRIPTION
**Pull request by tarekziade**
### Closes https://github.com/elastic/enterprise-search-team/issues/3285

A few changes in the `bulk` calls:

- Added `max_concurrency` to allow concurrent bulks
- Use a pass-through for `bulk` options to reduce friction in pass-through classes
- `get_size` stays in bytes and the logger can do its thing
- document classes

#### Pre-Review Checklist
- [X] Covered the changes with automated tests
- [X] Tested the changes locally

#### Changes Requiring Extra Attention

- [ ] Security-related changes (encryption, TLS, SSRF, etc)
- [ ] New external service dependencies added.

### CODE REVIEW

**Comment by seanstory** [comment]
is this tied to a ticket?

File: connectors/byoei.py [hunk 1]
```diff
@@ -34,10 +48,34 @@
 TIMESTAMP_FIELD = "_timestamp"
 
 
+def get_mb_size(ob):
```
Comment (moxarth-elastic) [inline connectors/byoei.py:51]: docstring missing

File: connectors/byoei.py [hunk 2]
```diff
@@ -195,6 +248,7 @@ async def get_docs(self, generator):
                         data.pop(TIMESTAMP_FIELD, None)
                         doc.update(data)
 
+                # logger.debug("Fetcher put")
```
Comment (moxarth-elastic) [inline connectors/byoei.py:251]: shall we remove this comment?

File: docs/CONFIG.md [hunk 3]
```diff
@@ -9,13 +9,15 @@ Configuration lives in [config.yml](../config.yml).
...
-  - `bulk_chunk_max_mem_size`: The max size in MB of a bulk request. When the next request being
-     prepared reaches that size, the query is emited even if `bulk_chunk_size` is not yet reached. Defaults to 5.
-  - `bulk_chunk_size`: The max size of the bulk operation to Elasticsearch. Defaults to 500.
+  - `bulk`: Options for the Bulk API calls behavior
+    - `display_every`: The number of docs between each counters display. Defaults to 100.
+    - `queue_max_size`: The max size of the bulk queue. Defaults to 1024.
+    - `queue_max_mem_size`: The max size in MB of the bulk queue. When it's reached, the next put
+       operation waits for the queue size to get under that limit. Defaults to 25.
+    - `chunk_max_mem_size`: The max size in MB of a bulk request. When the next request being
+       prepared reaches that size, the query is emited even if `bulk_chunk_size` is not yet reached. Defaults to 5.
```
Comment (praveen-elastic) [inline docs/CONFIG.md:18]: `bulk_chunk_size` -> `chunk_size`

File: docs/CONFIG.md [hunk 4]
```diff
@@ -9,13 +9,15 @@ Configuration lives in [config.yml](../config.yml).
...
-     operation waits for the queue size to get under that limit. Defaults to 25.
-  - `bulk_chunk_max_mem_size`: The max size in MB of a bulk request. When the next request being
-     prepared reaches that size, the query is emited even if `bulk_chunk_size` is not yet reached. Defaults to 5.
-  - `bulk_chunk_size`: The max size of the bulk operation to Elasticsearch. Defaults to 500.
+  - `bulk`: Options for the Bulk API calls behavior
+    - `display_every`: The number of docs between each counters display. Defaults to 100.
+    - `queue_max_size`: The max size of the bulk queue. Defaults to 1024.
+    - `queue_max_mem_size`: The max size in MB of the bulk queue. When it's reached, the next put
+       operation waits for the queue size to get under that limit. Defaults to 25.
+    - `chunk_max_mem_size`: The max size in MB of a bulk request. When the next request being
```
Comment (moxarth-elastic) [inline docs/CONFIG.md:17]: typo `emited` -> `emitted`

File: docs/CONFIG.md [hunk 5]
```diff
@@ -9,13 +9,15 @@ Configuration lives in [config.yml](../config.yml).
...
+       prepared reaches that size, the query is emited even if `bulk_chunk_size` is not yet reached. Defaults to 5.
+    - `chunk_size`: The max size of the bulk operation to Elasticsearch. Defaults to 500.
```
Comment (praveen-elastic) [inline docs/CONFIG.md:19]: The default value provided in utils is 100. Should we change it here ? https://github.com/elastic/connectors-python/blob/038f0daab4319f95640d21df30215b53100a1884/connectors/utils.py#L28

**Review by praveen-elastic — APPROVED** [review]

**Review by moxarth-elastic — APPROVED** [review]
LGTM👍

**Review by wangch079 — APPROVED** [review]

**Pull request merged**